analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,213 @@
+---
+title: "Find most focal recurrent copy number units"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Chante Bethell for ALSF CCDL
+date: 2020
+params:
+  is_ci: FALSE
+---
+
+This notebook defines the most focal recurrent copy number units by separating out entire chromsome arm losses and gains.
+
+## Usage
+
+This notebook is intended to be run from the command line with the following (assumes you are in the root directory of the repository):
+
+```
+Rscript -e "rmarkdown::render('analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+library(tidyverse)
+```
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+
+# Define a logical object for running in CI
+running_in_ci <- params$is_ci
+```
+
+### Read in files
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz"))
+
+# Read in the annotated CN file (without the UCSC data)
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz"))
+```
+
+## Define most focal units
+
+### First, we want find entire chromosome arm losses/gains.
+
+```{r}
+consensus_seg_cytoband_status_df <-
+  consensus_seg_cytoband_status_df %>%
+  select(chromosome_arm,
+         dominant_status,
+         cytoband,
+         Kids_First_Biospecimen_ID)
+
+arm_status_count <- consensus_seg_cytoband_status_df %>%
+  # Count the amount og times each arm has a specific `dominant_status` value
+  count(chromosome_arm, dominant_status) %>%
+  # Spread the data -- each row represents a unique chromosome arm
+  spread(dominant_status, n) %>%
+  select(-c("uncallable", "unstable")) %>%
+  replace_na(list(
+    gain = 0,
+    loss = 0,
+    neutral = 0
+  )) %>%
+  group_by(chromosome_arm) %>%
+  # The logic here is to define the arm status based on the majority of calls
+  # in the chromosome arm -- if the number of calls for a specific status is
+  # responsible for more than 50% of the total calls in that arm, then it is
```

Comment:
Technically, this is calling things when it is 49% or higher. e.g. if something is. 49.1% it will be called. I'd like to know how close these percentages are when something is called.

Reply:
The idea here was really to call an arm a loss/gain/neutral if that status is responsible for more than half of the calls in the arm.

_However_, the 0.49 value was included when I noticed that a lot of neutral dominant arm statuses and cytoband loss statuses (no gains) were in the results table.

That being said, I should probably raise the threshold back up to 0.5 (or higher) as I now notice instances where rows that should be neutral (same amount of gain and loss calls) are not called as neutral (with this logic).

I can wait to do this however, after I've calculated percentages for each call and we've determined an appropriate logic to make these calls.

Does this kind of answer your question?

Reply:
Well if you are looking for more than half, I think it makes sense to keep the logic consistent with that, so > 0.5 , partially so people don't get confused, but partially so that if there is an instance where the 0.01 difference matters, you are still all set.